pkg/storage/status_checker.go

```diff
@@ -191,41 +194,35 @@ func (sc *StatusChecker) processJob(
 		return fmt.Errorf("failed to get status: %v", err)
 	}
 
-	// Find active deals for the jobs
-	deals := sc.getActiveDeals(status, job)
-	if len(deals) == 0 {
-		fmt.Println("skipping adding deals")
+	// Find active activeDeals for the jobs
+	activeDeals := sc.checkActiveDeals(status, job)
+	if !activeDeals {
+		fmt.Println("skipping indexing cid")
 		return nil
 	}
 
-	if !sc.simulated {
-		fmt.Printf(
-			"not simulated: checking for duplicates: %s, %x \n",
-			pub, job.Cid)
-		deals, err = sc.removeDuplicateDeals(ctx, pub, deals)
-		// Don't fail if we can't remove duplicates, log it
-		// and continue to add deals
-		if err != nil {
-			fmt.Println("failed to remove duplicates: ", err)
-		}
+	cid, err := cid.Cast(job.Cid)
```

Comment:
interesting. was looking for this transformation the other day and could not find it